lib/discourse/single_sign_on.rb

```diff
@@ -12,7 +12,7 @@ class SingleSignOn
              :suppress_welcome_message].freeze
     NONCE_EXPIRY_TIME = 10.minutes
 
-    attr_accessor(*ACCESSORS, :sso_secret, :sso_url)
+    attr_accessor(*ACCESSORS)
 
     def self.sso_secret
       raise "sso_secret not implemented on class, be sure to set it on instance"
```

Comment:
Hmm I don't see the instance method declared elsewhere, so I wonder if it's confused the class method as a duplicate.

I can't actually find any specs covering this, so I tried constructing a quick test in the console:

```
require 'discourse/single_sign_on'
sso = Discourse::SingleSignOn.parse("query_string", "secret")
```
```
NoMethodError: undefined method `sso_secret=' for #<Discourse::SingleSignOn:0x0000000114c12798>

      sso.sso_secret = sso_secret if sso_secret
         ^^^^^^^^^^^^^
Did you mean?  sso_secret
```

This is obviously not an exhaustive test, so we should either add specs, or  definitely do a manual test of the functionality. Hmm, but I can't find any documentation about this.. I wonder if it's even used...

Reply:
`±` To fix: the class methods (`self.sso_secret` etc) seem unnecessary, so I'd consider removing them instead. And why not move `:sso_secret, :sso_url` to the `ACCESSORS` list.. 

But still I'd like to confirm if it's worth it first.

Reply:
Update: this whole file is unused and so I've [scheduled for removal](https://github.com/openfoodfoundation/openfoodnetwork/issues/12355).

For now, I suggest we add `rubocop:disable` for this whole file and move on.